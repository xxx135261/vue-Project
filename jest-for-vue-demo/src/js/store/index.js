import Vuex from 'vuex'
import Vue from 'vue'
import axios from 'axios'

Vue.use(Vuex);

export const state = {
    msgList: [],
    maxListSize: 1000,
    echartConfig: {
        tooltip: {
            trigger: 'item',
            backgroundColor:'#528DFF',
            color:'#fff',
            formatter: "{a} <br/>{b} : {c} ({d}%)"
        },
        toolbox: {
            show: true,
            feature: {
                dataView: {
                    readOnly: true,
                    iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                },
                restore: {
                	iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                },
                saveAsImage: {
                    name: '行业配置',
                    iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                }
            },
            right: "16px"
        },
        legend: {
            orient: 'vertical',
            left: 'left',
            data: ['农业', '计算机', '其他']
        },
        series: [{
            type: 'pie',
            radius: '55%',
            center: ['50%', '68%'],
            data: [
                {value: 35, name: '农业'},
                {value: 10, name: '计算机'},
                {value: 55, name: '其他'}
            ]
        }],
        grid: {
            left: 'center'
        }
    }
}


export const getters = {
    msgAmount: state => state.msgList.length
}

export const actions = {
    getMsgFromRemote: ({commit}) => {
        axios
            .get('https://www.easy-mock.com/mock/5b765aef4d2b8f332fda9656/msgpool/mock-msg')
            .then(resp => {
                commit('pushMsg', {
                    msg: resp.data
                })
            })
    }
}

export const mutations = {
    pushMsg: (state, payload) => {
        let msgList = state.msgList, currLen = msgList.length;
        if (currLen === state.maxListSize) {
            msgList.splice(0, 1);
        }
        msgList.push(payload.msg);
    }
}

export default new Vuex.Store({
    state,
    getters,
    mutations,
    actions
});