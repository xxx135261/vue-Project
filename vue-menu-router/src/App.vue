<template>
  <div id="app" class="full">
    <router-view />
  </div>
</template>
<script>
import {getRouterMenu} from '@/router'
export default {
	name: 'App',
	created(){
		this.initMenuList();
	},
  methods: {
		initMenuList(){
			getRouterMenu().then(data => {
				if(data.menus.length>0){
					this.$store.commit('setMenulist', data.menus);
					this.$router.addRoutes(data.router)
				}else{
					this.$message({
						showClose: true,
						message: '菜单获取失败！',
						type: 'warning'
					});
					this.$router.push({path:'/'})
				}
			})
		},

  },
}

</script>
<style>
body{
  margin: 0;padding: 0;
}
.full{
	position: fixed;
	left: 0;
	right: 0;
	top: 0;
	bottom: 0;
}

</style>
