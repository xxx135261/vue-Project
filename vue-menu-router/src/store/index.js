import Vue from 'vue'
import Vuex from 'vuex'
Vue.use(Vuex)
const state = {
  menulist:[]
}
const	getters = {
	menulist: state => {
		return state.menulist;
	}
}
const	mutations = {
	setMenulist:(state,payLoad) => {
		state.menulist = payLoad
	}
}
export default new Vuex.Store({
	state,
	getters,
	mutations
})
