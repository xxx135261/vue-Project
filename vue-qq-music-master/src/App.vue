<template>
  <div id="app">
    <nav-header></nav-header>
    <keep-alive>
      <router-view/>
    </keep-alive>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import NavHeader from './components/NavHeader.vue';

@Component({
    components: {
        NavHeader
    }
})
export default class App extends Vue {}
</script>

<style lang="scss">
@import './style/_var';
@import './style/reset';
body {
    background: $body-background;
}
.noscroll {
    overflow: hidden;
}
</style>