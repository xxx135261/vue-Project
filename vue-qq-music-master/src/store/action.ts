import {
    SET_HISTORY_KEY,
    SAVE_LOVE,
    GET_SONG,
    SHOW_PLAYER,
    CANCEL_PLAYER,
    SET_PLAYING_STATE
} from './mutation-types';
import { ActionTree, Action } from 'vuex';

// 存储历史记录
const setLocalStorageData: Action<any, any> = ({ commit }, payload) => {
    commit(SET_HISTORY_KEY, payload);
};

const getSong: Action<any, any> = ({ commit }, payload) => {
    commit(GET_SONG, payload);
};

const showPlayer: Action<any, any> = ({ commit }) => {
    commit(SHOW_PLAYER);
};
// 存储收藏的歌曲（触发mutation中的SAVE_LOVE事件
const setLocalStorageData2: Action<any, any> = ({ commit }, payload) => {
    commit(SAVE_LOVE, payload);
};

const cancelPlayer: Action<any, any> = ({ commit }) => {
    commit(CANCEL_PLAYER);
};

const setPlayState: Action<any, any> = ({ commit }, payload) => {
    commit(SET_PLAYING_STATE, payload);
};

const actions: ActionTree<any, any> = {
    setLocalStorageData,
    setLocalStorageData2,
    getSong,
    showPlayer,
    cancelPlayer,
    setPlayState
};
export default actions;
