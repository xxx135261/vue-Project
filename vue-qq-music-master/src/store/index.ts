import Vue from 'vue';
import Vuex from 'vuex';
import mutations from './mutations';
import actions from './action';
Vue.use(Vuex);

export const state = {
    isShowPlayer: false, // 是否显示 player
    song: {}, // 歌的信息
    artist: '', // 歌的作者
    isPlay: false, // 是否播放
    msg: 'hello world!'
};
export default new Vuex.Store({
    state,
    actions,
    mutations
});
