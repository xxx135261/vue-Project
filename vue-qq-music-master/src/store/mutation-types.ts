// 缓存搜索历史
export const SET_HISTORY_KEY = 'SET_HISTORY_KEY';
// 缓存收藏的歌曲
export const SAVE_LOVE = 'SAVE_LOVE';
// 获取歌曲
export const GET_SONG = 'GET_SONG';
// 显示 player
export const SHOW_PLAYER = 'SHOW_PLAYER';
// 不显示 player
export const CANCEL_PLAYER = 'CANCEL_PLAYER';
// 播放状态（true/false)
export const SET_PLAYING_STATE = 'SET_PLAYING_STATE';
