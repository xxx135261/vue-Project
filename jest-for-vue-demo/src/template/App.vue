<template>
    <div class="container-layout">
        <h1 :class="msgClass">{{msg}}</h1>
        <div class="toggle-btn inline right" @click="togglePoolSwitch">{{operator}}</div>
        <div class="pool-container">
            <MessagePool ref="pool"></MessagePool>
        </div>
        <div v-echarts config="echartConfig" ref="attribute"
            style="width: 100%;height: 312px;">
        </div>
    </div>
</template>
<script>
import MessagePool from './MessagePool';
import '../js/echarts/vue-echarts.js';
export default {
    name: 'App',
    components: { MessagePool },
    data() {
        return {
            operator: '暂停',
            activated:true,
        }
    },
     activated() {
            this.activated = true;
            debugger
            if (this.chart && this.chart.length > 0) {
                this.chart.forEach(function(val) {
                    setTimeout(val.resize, 0);
                })
            }
        },
    deactivated() {
        this.activated = false;
    },
    computed: {
        msgClass() {
            return {
                'font-active': this.operator === '暂停',
                'font-inactive': this.operator === '继续',
                'font-family': true,
                'inline': true,
                "info": true
            }
        },
        msg() {
            return 'Amount of message: ' + this.$store.getters.msgAmount;
        },
        echartConfig(){
            return this.$store.state.echartConfig;
        }
    },
    methods: {
        togglePoolSwitch() {
            let operator = this.operator;
            let pool = this.$refs['pool'];
            operator === '暂停' ? pool.stop() : pool.resume();
            operator === '暂停' ? this.operator = '继续' : this.operator = '暂停';
        }
    }
}
</script>
<style>
    body {
        background: #2d2d47
    }
    .font-active {
        color: #dddddd
    }
    .font-inactive {
        color: #6f6983
    }
    .font-family {
        font-family: Helvetica, Courier, monospace, Arial, "Microsoft Yahei", "微软雅黑";
        font-weight: lighter
    }
    .inline {
        display: inline-block;
    }
    .info {
        margin-left: calc(50% - 190px);
    }
    .right {
        float: right;
        margin-top: 10px;
        margin-right: 10px;
    }
    .pool-container {
        margin: 0 calc(50% - 400px)
    }
    .toggle-btn {
        width: 50px;
        height: 50px;
        background: #4e6b9f;
        -moz-border-radius: 25px;
        -webkit-border-radius: 25px;
        border-radius: 25px;
        line-height: 50px;
        text-align: center;
        color: white;
        cursor: pointer;
    }
    .container-layout {
        width: 100%;
        height: 100%;
    }
</style>