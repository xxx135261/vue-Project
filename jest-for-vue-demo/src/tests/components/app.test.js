import App from './../../template/App.vue'
// 通过vue-test-utils提供的mount方法来挂载组件，创建包裹器和Vue实例
import {mount} from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import {state, getters} from './../../js/store'

Vue.use(Vuex);

const msgValue = 'Amount of message: 0';

describe('App.test.js', () => {
    let wrapper;
    let store;
    let GET_MSG_FROM_REMOTE, PUSH_MSG;
  
    beforeEach(() => {
      // 通过jest.fn()生成一个mock函数
      GET_MSG_FROM_REMOTE = jest.fn();
      PUSH_MSG = jest.fn();
      
      store = new Vuex.Store({
        state,
        getters,
        mutations: {
          pushMsg: PUSH_MSG
        },
        actions: {
            getMsgFromRemote: GET_MSG_FROM_REMOTE
        }        
      });
      wrapper = mount(App, {
        mocks: {
          $store: store
        }
      });
    })
  
    test(`equals messages to ${msgValue}`, () => {
      expect(wrapper.vm.msg).toEqual(msgValue);
    })

    test('has expected html', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    })

    describe('test toggle button', () => {
        test('equals button default content to "暂停"', () => {
          expect(wrapper.find('.toggle-btn').element.textContent).toEqual('暂停');
        })

        test('test button content when clicked', () => {
          let btnWrapper = wrapper.find('.toggle-btn');
          expect(btnWrapper.element.textContent).toEqual('暂停');
          btnWrapper.trigger('click');
          expect(btnWrapper.element.textContent).toEqual('继续');
          btnWrapper.trigger('click');
          expect(btnWrapper.element.textContent).toEqual('暂停');
        })
    })

    /* structure and style testing */
    describe('test h1 classes', () => {
      test('element h1 default classes contain font-active', () => {
          let h1Wrapper = wrapper.find('h1');
          expect(h1Wrapper.classes()).toContain('font-active');
      })

      test('element h1 classes changed by clicking button', () => {
          let h1Wrapper = wrapper.find('h1');
          let btnWrapper = wrapper.find('.toggle-btn');
          expect(h1Wrapper.classes()).toContain('font-active');
          btnWrapper.trigger('click');
          expect(h1Wrapper.classes()).toContain('font-inactive');
      })
    })

    describe('test custom event', () => {
      let updateHandler = jest.fn(), spied;
      beforeEach(() => {
        // replace with a mock function
        wrapper.setMethods({
          poolUpdate: updateHandler
        });
        // spy on poolUpdate()
        spied = jest.spyOn(wrapper.vm, 'poolUpdate');
      })

      test('custom event emitted calls update handler', () => {
        jest.useFakeTimers();
        setTimeout(() => {
          expect(spied).toHaveBeenCalled();
        }, 2000);
      })
    })
  })