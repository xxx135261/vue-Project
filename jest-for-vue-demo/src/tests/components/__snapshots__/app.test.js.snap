// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has expected html 1`] = `
<div
  class="container-layout"
>
  <h1
    class="font-active font-family inline info"
  >
    Amount of message: 0
  </h1>
   
  <div
    class="toggle-btn inline right"
  >
    暂停
  </div>
   
  <div
    class="pool-container"
  >
    <div
      class="pool"
    >
      <ul />
    </div>
  </div>
</div>
`;
